stetho/src/main/java/com/facebook/stetho/inspector/network/BasePrettyPrinterFactory.java

```diff
@@ -0,0 +1,35 @@
+package com.facebook.stetho.inspector.network;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+
+import com.facebook.stetho.common.Util;
+import com.facebook.stetho.inspector.network.AsyncPrettyPrinter;
+import com.facebook.stetho.inspector.network.AsyncPrettyPrinterFactory;
+
+/**
+ * Implementation of {@link AsyncPrettyPrinterFactory} which generates asynchronous pretty printers
+ * that allows callers to pretty print responses with specific header names
+ */
+public abstract class BasePrettyPrinterFactory implements AsyncPrettyPrinterFactory {
+
+  @Override
+  public AsyncPrettyPrinter getInstance(String headerName, String headerValue) {
+    //TODO: async download schema
```

Comment:
This can't be landed as-is.  Are you intending to clean this up before the PR is merged?